pkg/csi/controller_server.go

```diff
@@ -89,11 +116,31 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 			Name:      req.Name,
 		},
 		Spec: corev1.PersistentVolumeClaimSpec{
-			StorageClassName: pointer.StringPtr(cs.hostStorageClass),
-			AccessModes:      []corev1.PersistentVolumeAccessMode{corev1.ReadWriteMany},
-			VolumeMode:       &volumeMode,
+			AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteMany},
+			VolumeMode:  &volumeMode,
 		},
 	}
+
+	/*
+	 * Let's handle SC
+	 * SC define > controller define
+	 *
+	 * Checking mechanism:
+	 * 1. If guest cluster is not be allowed to list host cluster StorageClass, just continue.
+	 * 2. If guest cluster can list host cluster StorageClass, check it.
+	 */
+	if val, exists := volumeParameters[paramHostSC]; exists {
+		//If StorageClass has `hostStorageClass` parameter, check whether it is valid or not.
+		if valid, err := cs.isStorageClassValid(targetSC); !valid {
+			return nil, status.Error(codes.Aborted, err.Error())
```

Comment:
Prefer to use codes.Internal

Reply:
updated with the whole above suggestions, thanks!!